spec/requests/saved_pages_request_spec.rb

```diff
@@ -0,0 +1,45 @@
+RSpec.describe "SavedPages", type: :request do
+  context "when receiving an unauthenticated request" do
+    it "returns unauthorised for GET /api/saved_pages" do
+      get saved_pages_path
+
+      expect(response).to have_http_status(:unauthorized)
+    end
+
+  end
+
+  context "when receiving an authenticated request" do
+    before do
+      stub_oidc_discovery
+
+      # rubocop:disable RSpec/AnyInstance
+      allow_any_instance_of(OidcClient).to receive(:tokens!).and_return({ access_token: "access-token", refresh_token: "refresh-token" })
+      # rubocop:enable RSpec/AnyInstance
+
+      FactoryBot.create(:oidc_user)
+    end
+
+    let(:headers) { { "Content-Type" => "application/json", "GOVUK-Account-Session" => session_identifier } }
+    let(:session_identifier) { placeholder_govuk_account_session }
+    let(:user) { OidcUser.find_by(sub: "user-id") }
+
+    describe "GET /api/saved_pages" do
+      it "returns an empty array if there are no saved pages" do
+        get saved_pages_path, headers: headers
+
+        expect(response).to have_http_status(:ok)
+        expect(JSON.parse(response.body)["saved_pages"]).to eq([])
```

Comment:
Not specifically related to this PR so feel free to ignore this: we use `JSON.parse(response.body)` a lot. It would be nice to have a `response_body` method or a `let(:response_body) { JSON.parse(response.body) }` somewhere that we could use in all our tests.

Reply:
I think you are right, perhaps we can have a tidy up PR after this?

Reply:
Sounds good